src/common/regex.cpp

```diff
@@ -690,4 +690,39 @@ int wxRegEx::Replace(wxString *pattern,
     return m_impl->Replace(pattern, replacement, maxMatches);
 }
 
+wxString wxRegEx::Escape(const wxString& str)
+{
+    wxString strEscaped;
+
```

Comment:
Should call `strEscaped.reserve(str.length())`, this doesn't cost anything and can avoid an extra memory allocation or two.

Reply:
May be `strEscaped.reserve(str.length() * 2)` even better. This is the maximum possible length of the escaped string. And then call `strEscaped.Shrink()` after the loop.

Reply:
Probably not catastrophic here, but generally speaking `*2` is a lot. Hopefully nobody will be using multimegabyte regexes however...

OTOH you wanted to call `Shrink()` but forgot to do it. Oversight?

Reply:
indeed, I forgot to do it, I am doing it now.